e2e/utils.go

```diff
@@ -82,6 +90,150 @@ var (
+func validateOmapCount(f *framework.Framework, count int, driver, pool, mode string) {
+	var (
+		radosLsCmd           string
+		radosLsCmdFilter     string
+		radosLsKeysCmd       string
+		radosLsKeysCmdFilter string
+	)
+	switch strings.ToLower(mode) {
+	case volumesType:
+		switch strings.ToLower(driver) {
+		case cephfsType:
+			radosLsCmd = fmt.Sprintf("rados ls --pool=%s --namespace csi", pool)
+			radosLsCmdFilter = fmt.Sprintf("rados ls --pool=%s --namespace csi | grep -v default | grep -c ^csi.volume.", pool)
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.volumes.default --pool=%s --namespace csi", pool)
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.volumes.default --pool=%s --namespace csi | wc -l", pool)
+		case rbdType:
+			radosLsCmd = fmt.Sprintf("rados ls %s", rbdOptions(pool))
+			radosLsCmdFilter = fmt.Sprintf("rados ls %s | grep -v default | grep -c ^csi.volume.", rbdOptions(pool))
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.volumes.default %s", rbdOptions(pool))
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.volumes.default %s | wc -l", rbdOptions(pool))
+		default:
+			e2elog.Failf("unknown driver type: %s", driver)
+		}
+	case snapsType:
+		switch strings.ToLower(driver) {
+		case cephfsType:
+			radosLsCmd = fmt.Sprintf("rados ls --pool=%s --namespace csi", pool)
+			radosLsCmdFilter = fmt.Sprintf("rados ls --pool=%s --namespace csi | grep -v default | grep -c ^csi.snap.",
+				pool)
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.snaps.default --pool=%s --namespace csi", pool)
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.snaps.default --pool=%s --namespace csi | wc -l",
+				pool)
+		case rbdType:
+			radosLsCmd = fmt.Sprintf("rados ls %s", rbdOptions(pool))
+			radosLsCmdFilter = fmt.Sprintf("rados ls %s | grep -v default | grep -c ^csi.snap.", rbdOptions(pool))
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.snaps.default %s", rbdOptions(pool))
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.snaps.default %s | wc -l", rbdOptions(pool))
+		default:
+			e2elog.Failf("unknown driver type: %s", driver)
+		}
+	default:
+		e2elog.Failf("unknown mode type: %s", mode)
+	}
+	stdOut, stdErr, err := execCommandInToolBoxPod(f, radosLsCmdFilter, rookNamespace)
+	if err != nil || stdErr != "" {
+		if !strings.Contains(err.Error(), exitOneErr) {
+			e2elog.Failf("failed to execute rados command '%s' : err=%v stdErr=%s", radosLsCmdFilter, err, stdErr)
+		}
+	}
+	var err0 error
+	err = compareStdoutWithCount(stdOut, count)
+	if err != nil {
+		if !strings.Contains(err.Error(), "expected omap object count") {
+			e2elog.Failf("%v", err)
+		}
+		stdOut, stdErr, err0 = execCommandInToolBoxPod(f, radosLsCmd, rookNamespace)
+		if err0 == nil {
+			e2elog.Logf("additional debug info: rados ls command output: %s, stdErr: %s, err: %v",
+				stdOut, stdErr, err0)
+		}
+		e2elog.Failf("%v", err)
+	}
```

Comment:
```suggestion
err = compareStdoutWithCount(stdOut, count)
if err != nil {
	e2elog.Failf("%v. command executed %s and command output is %s", err,radosLsCmd,stdOut)
}
```
can be simplified as above?

Reply:
No, we need to execute the command again, it was you who asked for `rados` command output.

Reply:
got it this is without filter to list all the omap entires. should be fine :+1: